src/App.tsx

```diff
@@ -15,47 +16,140 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      return (isReversed)
+        ? [...visibleGoods].sort((a, b) => b.localeCompare(a))
+        : [...visibleGoods].sort((a, b) => a.localeCompare(b));
+    case SortType.LENGTH:
+      return (isReversed)
+        ? [...visibleGoods].sort((a, b) => b.length - a.length)
+        : [...visibleGoods].sort((a, b) => a.length - b.length);
+    default:
+      return (isReversed)
+        ? [...visibleGoods].reverse()
+        : [...visibleGoods];
+  }
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState([...goodsFromServer]);
+  const [activeSort, setActiveSort] = useState(SortType.NONE);
+  const [activeReset, setActiveReset] = useState(false);
```

Comment:
```suggestion
```
You don't need this in the state since reset is active when sort is not NONE or isReversed is true. There is no need to hold in the state data which value depends on other state values only